autogen.sh

```diff
@@ -824,7 +824,6 @@ if [ $do_bindings = "yes" ] ; then
 	# Top-level files
 	( cd src/binding/fortran/use_mpi_f08 && chmod a+x ./buildiface && ./buildiface )
         # generate src/binding/fortran/use_mpi_f08/wrappers_c/...
-        $PYTHON maint/gen_binding_f08.py
```

Comment:
So this introduces a `python3` dependency on tarball builds, not just developer builds.

Reply:
I don't have a strong feeling either way. We already require for ourselves so it doesn't impact me. I don't know how users will feel about it.

Is there a strong need for it? We use it in the MPI Forum I assume because it's easier than not using it, but it could probably be written without it.

That being said (and I'm not a Python expert), Python 3 is how old now?

Reply:
Yes. Python 2 has been officially deprecated for a while now, so I expect most systems have python 3 in one form or another.

Reply:
I just don't understand why people hate Perl, it won't have any such issues...

Reply:
Please correct me if I am wrong, but tarball builds are still quite bare in their dependencies. Adding Python, in any form, is a significant change. In my mind, it's
```
sh
C
C++ (optional)
Fortran (optional)
make
```
Am I missing anything?

Reply:
Yes, It is a significant change. It is adding --
```
   Python 3 (optional, required to build f08)
```
Just to be clear, this is a solution so the f08 build can correctly --

1. properly support `real128`
2. properly support `--disable-romio`

Currently, users have to 

1. manually delete the `real128` code if their system doesn't support `real128`
2. can't build without romio or they won't be able to link any Fortran program

The potential benefit, in addition to fixing current issues,

1. Smaller distribution tarballs
2. Potential to simplify configure as we may be able to move some complexities from configure into Python and let configure just do feature detection.

And the con side --

1. User will need install Python 3 if they need build Fortran with f08

Reply:
For the benefit of understanding, how much more complex would this be to do with conditional compilation vs. generation? What I'm getting at is, how many files are affected by the configuration? Could we pre-generate multiple versions in` autogen.sh` and use `AM_CONDITIONAL` to build the "right" ones? I realize this means `autogen.sh` still has to do some unnecessary work, but the new buildiface scripts seem way faster that the old ones, so that's less of a concern in my mind.

Reply:
I do want to avoid generating the full cross-product of
```
real128+romio
no real128+romio
real128+no romio
no real128+no romio
```
if possible.

Reply:
It is not complex to achieve, but it is not elegant -- any newly discovered or developed similar issue will exponentially grow the solution.

If we are really driven to avoid Python dependency, I think we can --
1. move `mpi_f08_types.f90` to `mpi_f08_types.f90.in` and let configure to replace `real128` into `real64`.
2. Build empty stubs with `--disable-romio`

I am really motivated by the prospect of reducing `xxx.in` files and simplify configure complexities. How about let's first discuss the "con"-side of having Python dependency. What is the practical impact?

Reply:
> I am really motivated by the prospect of reducing `xxx.in` files and simplify configure complexities. How about let's first discuss the "con"-side of having Python dependency. What is the practical impact?

Most likely a lot of confused users asking why `./configure` failed 😄. This can _hopefully_ be overcome with a good error message and the passage of time. Package maintainers can be notified in a release announcement, so I'm less concerned about them.

And just to be clear, I'm not really thinking about Python 2 vs. 3, just Python in general.

Reply:
A point of reference: PETSc is one project that requires Python in their `configure` script.
```
if sys.version_info < (2,6) or (sys.version_info >= (3,0) and sys.version_info < (3,4)):
  print('************************************************************************')
  print('*      Python version 2.6+ or 3.4+ is required to run ./configure      *')
  print('*         Try: "python2.7 ./configure" or "python3 ./configure"        *')
  print('************************************************************************')
  sys.exit(4)
```

Reply:
That's a Python script. How would they print that message if "Python" is not available?

Reply:
Oh, their configure apparently is a python script

Reply:
Right. I'll reach out to those folks to see if they can share any experience/headaches from using Python in configure.

Reply:
Are they using Meson https://github.com/mesonbuild/meson?

Reply:
Found a whole page of python build tools: https://wiki.python.org/moin/ConfigurationAndBuildTools